projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		if (S3_detected) {
+			flushBuff(gBuff, sizeof(gBuff));
+			sprintf((char*)gBuff, "%ld, ", SensorReadoutPPB(ADI_CFG_I2C_SENSOR3_ADDR));
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		if (S4_detected) {
+			flushBuff(gBuff, sizeof(gBuff));
+			sprintf((char*)gBuff, "%ld, ", SensorReadoutPPB(ADI_CFG_I2C_SENSOR4_ADDR));
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		UART_TX((const char*)TXbuff);
+		break;
+
+	case WATER: {
+		if(initialcycle) {
+			//Do initial read of water quality board transmit buffer
+			if (S1_detected) {
+				eSensorResult = pGasSensor->openWithAddr(ADI_CFG_I2C_SENSOR1_ADDR);
+				if(eSensorResult != SENSOR_ERROR_NONE) UART_TX("Error opening sensor 1!" _EOS);
+				else UART_TX("Sensor Site: 1");
+			} else if (S2_detected) {
+				eSensorResult = pGasSensor->openWithAddr(ADI_CFG_I2C_SENSOR2_ADDR);
+				if(eSensorResult != SENSOR_ERROR_NONE) UART_TX("Error opening sensor 2!" _EOS);
+				else UART_TX("Sensor Site: 2");
+			} else if (S3_detected) {
+				eSensorResult = pGasSensor->openWithAddr(ADI_CFG_I2C_SENSOR3_ADDR);
+				if(eSensorResult != SENSOR_ERROR_NONE) UART_TX("Error opening sensor 3!" _EOS);
+				else UART_TX("Sensor Site: 3");
+			} else if (S4_detected) {
+				eSensorResult = pGasSensor->openWithAddr(ADI_CFG_I2C_SENSOR4_ADDR);
+				if(eSensorResult != SENSOR_ERROR_NONE) UART_TX("Error opening sensor 4!" _EOS);
+				else UART_TX("Sensor Site: 1");
+			} else {
+				UART_TX("Error: No board found!" _EOS);
+				break;
+			}
+			numbytes = 0;
+			pGasSensor->I2CReadWrite(READ, BYTES_TO_READ, (uint8_t *)&numbytes,
+						 1);//numbytes can be 0 to 255, but there can be more than 255 bytes in slave buffer
+			delay_ms(5);
+			while(numbytes > 0) {
+				pGasSensor->I2CReadWrite(READ, 0x63, &Slave_Rx_Buffer[0], numbytes);
+				delay_ms(10);
+				for(Slave_Rx_Index = 0; Slave_Rx_Index<numbytes; Slave_Rx_Index++) {
+					tempChar = Slave_Rx_Buffer[Slave_Rx_Index];
+					Slave_Rx_Buffer[Slave_Rx_Index] = '\0';
+					if(tempChar != '~' && tempChar != '\0') {
+						UART_TX_DIR((const char *)&tempChar);
+					}
+				}
+				pGasSensor->I2CReadWrite(READ, BYTES_TO_READ, (uint8_t *)&numbytes, 1);
+				delay_ms(5);
+			} //end while data available loop
+			initialcycle = 0;
+		}
+		/*******************Streaming data from UART to I2C and back********************/
+		while(true) { //main water quality loop
+			if (cmdReceived) {
+				//Command Received. Check if we need to switch sensor, otherwise send command to sensor board.
+				cmdReceived = 0;
+				cmdLen = strlen((const char*)cmdInString);
+				cmdInString[cmdLen] = _CR; //add carriage return back in to send to slave
+				if(strncmp((const char *)cmdInString, "switchsensor", 11) == 0) {
+					tempChar = cmdInString[cmdLen - 1];//character before the enter key
+					UART_TX_DIR("Switching to site ");
+					UART_TX((const char *)&tempChar);
+					if (tempChar == '1' && S1_detected) {
```

Comment:
similarly, related to comment https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/12/commits/4f5694854b122faead93c95e10cae42d6207dad4#r243179682  ; this can be a `for` block

Reply:
This should be possible as long as I'm careful about the character comparisons where I just used character literals before.